client/components/app-promo/index.jsx

```diff
@@ -32,37 +37,28 @@ export default React.createClass( {
 			show_promo = false;
 		}
 
-		const chromeRE = /\bCrOS\b/;
-		if ( chromeRE.test( navigator.userAgent ) ) {
+		if ( isChromeOS() ) {
 			show_promo = false;
 		}
 
-		const promo_options = [
-			{ promo_code: 'a0001', message: 'WordPress.com your way  — desktop app now available for Mac, Windows, and Linux.' },
-			{ promo_code: 'a0002', message: 'Get WordPress.com app for your desktop.' },
-			{ promo_code: 'a0003', message: 'WordPress.com app now available for desktop.' },
-			{ promo_code: 'a0005', message: 'WordPress.com at your fingertips — download app for desktop.' }
-		];
-
-		const item = promo_options[ Math.floor( Math.random() * promo_options.length ) ];
 		return {
-			promo_item: item,
+			promo_item: getRandomPromo(),
```

Comment:
if we move this into `mapStateToProps` or somewhere in `connect()` then we won't have to stub it out in tests. it's just a prop for the component and we can test it like this…

```js
const wrapper = shallow( <AppPromo promoItem="just a test" /> );
expect( wrapper.find( whatever ) ).to.be( 'just a test' );
```